sympy/core/mul.py

```diff
@@ -172,9 +172,14 @@ def flatten(cls, seq):
         """
 
         from sympy.calculus.util import AccumBounds
+        from sympy import Poly
         rv = None
         if len(seq) == 2:
```

Comment:
I thought I left a message earlier today, but don't see it now. I don't think the change should be happening here in the 2-arg handling portion of flatten. I'm not even sure it should happen in flatten at all (but I could be wrong about that).